spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/jmx/JmxEndpointExporter.java

```diff
@@ -89,6 +97,7 @@ private ObjectName register(ExposableJmxEndpoint endpoint) {
 		try {
 			ObjectName name = this.objectNameFactory.getObjectName(endpoint);
 			EndpointMBean mbean = new EndpointMBean(this.responseMapper, endpoint);
+			mbean.setBeanClassLoader(this.classLoader);
```

Comment:
I considered having the class loader as a constructor argument, but for me the class loader is optional for the `EndpointMBean`.